### PR TITLE
fix: bumping version should commit a changed package-lock.json

### PR DESCRIPTION
**Pull request by trentm**
https://github.com/elastic/opbeans-node/pull/72 added a package-lock.json to this repo to have repeatable versions.

However, the `.ci/bump-version.sh` that is run to bump the "elastic-apm-node" dependency version when a new release of the Node.js APM agent is tagged *does not commit changes to package-lock.json*.   The result is that we have a commit to the repo that *says* it has updated the elastic-apm-node dep (e.g. https://github.com/elastic/opbeans-node/commit/6ab7eba569a3b7126b1e65617320be888aaff893), but it won't actually use that newer version of the agent because package-lock.json hasn't been updated.

https://github.com/elastic/opbeans-node/pull/73 enabled dependabot on this repo. *That* will eventually create a PR to update the package-lock file. For example, in this case: https://github.com/elastic/opbeans-node/pull/146

This PR updates ".ci/bump-version.sh" to update the package-lock file. However, there is a potential issue. AFAIK there is no way to ask `npm install some-package ...` to update *only the part of package-lock.json that is relevant to just the given "some-package" argument*. Notice that the "es-abstract" dependency is updated in the PR below. This means that when a new apm-agent-nodejs.git release tag is made, we will get a commit to this repo that updates to the latest of all transitive deps. That is possibly a slight security concern.

# An alternative

An alternative would be to *not* use ".ci/bump-version.sh" and instead rely on the dependabot PR to update opbeans-node to the latest Node.js APM agent. This would mean dropping this section from the apm-agent-nodejs Jenkinsfile:
https://github.com/elastic/apm-agent-nodejs/blob/76bc0740c7b8c9140a6f58a0851767ea11970dcf/.ci/Jenkinsfile#L284-L302
It would mean we would not create tags on the opbeans-node repo. Are those needed?

Which do you prefer?
1. Get bump-version.sh to update package-lock.json, or
2. stop using bump-version.sh and rely instead on dependabot creating a PR that we can review before merging.

### CODE REVIEW

File: .ci/bump-version.sh
Comment (astorm) [inline .ci/bump-version.sh:14]: Just noting we've added  `--ignore-scripts` here.  I'm presuming this is related to discussion around SLSA/Supply Chain mitigation.

File: .ci/bump-version.sh
Comment (astorm) [inline .ci/bump-version.sh:19]: Question: Where does `Dockerfilebck` come from?

**Review by astorm — COMMENTED** [review]
Re:


> 1. Get bump-version.sh to update package-lock.json, or
> 2. stop using bump-version.sh and rely instead on dependabot creating a PR that we can review before merging.

No strong opinions here -- happy to defer to the opinion of folks more familiar with why bump-version.sh exists.

File: .ci/bump-version.sh
Comment (trentm) [reply .ci/bump-version.sh:19]: From the `sed -ibck ...` option:

```
     -I extension
             Edit files in-place, saving backups with the specified extension.  ...
             That can lead to unexpected
             results in many cases of in-place editing, where using -i is desired.

     -i extension
             Edit files in-place similarly to -I, but treat each file independently from other
             files. ...
```

File: .ci/bump-version.sh
Comment (trentm) [reply .ci/bump-version.sh:14]: Partially related to the SLSA discussions, but also to avoid this current package.json "scripts" entry:

```
    "postinstall": "if [ \"$NODE_ENV\" != production ]; then npm run client-update; fi",
```

which I think may be adding a lot of unnecessary time to this "bump-version.sh" run.

**Comment by trentm** [comment]
Note that the calling of "bump-version.sh", etc. from the apm-agent-nodejs.git Jenkinsfile was added in https://github.com/elastic/apm-agent-nodejs/pull/1553

**Comment by v1v** [comment]
Thanks for raising this, I'm not inclined to diverge from the original pattern. Although you raised a good point regarding `create tags on the opbeans-xxx repos`, maybe this is something we can discuss with the other APM agent teams and if agreed, then we can remove the label generation for all. Then, using dependabot won't cause any mismatch.

What do you think?

**Comment by v1v** [comment]
Thanks for raising this, I'm not inclined to diverge from the original pattern. Although you raised a good point regarding `create tags on the opbeans-xxx repos`, maybe this is something we can discuss with the other APM agent teams and if agreed, then we can remove the label generation for all. Then, using dependabot won't cause any mismatch.

What do you think?

**Comment by trentm** [comment]
> I'm not inclined to diverge from the original pattern.

Why is that? For some sense of consistency across the opbeans-\* repos?

Definitely worthwhile bringing up with the other APM agent teams, yes. However, I'm not necessarily sold that all the opbeans-\* repos need to use the same mechanism for updating their APM agent deps... beyond a suggestion/rule that they *do* update somehow.

**Comment by trentm** [comment]
A limitation of relying on dependabot (at least for opbeans-node) is that it would not handle updating the LABEL directive in the Dockerfile:

```
LABEL \
    org.label-schema.schema-version="1.0" \
    org.label-schema.vendor="Elastic" \
    org.label-schema.name="opbeans-node" \
    org.label-schema.version="3.31.0" \
    org.label-schema.url="https://hub.docker.com/r/opbeans/opbeans-node" \
    org.label-schema.vcs-url="https://github.com/elastic/opbeans-node" \
    org.label-schema.license="MIT"
```

(I now realize perhaps that is what you meant by your "then we can remove the label generation for all" above. :)
I'm curious if that docker label is used for anything.

(BTW, I noticed that opbeans-frontend is not getting updated with the latest `@elastic/apm-rum` releases.)

**Comment by trentm** [comment]
> (BTW, I noticed that opbeans-frontend is not getting updated with the latest `@elastic/apm-rum` releases.)

https://github.com/elastic/apm-agent-rum-js/issues/1191 opened for that.

**Comment by v1v** [comment]
> Why is that? For some sense of consistency across the opbeans-* repos?

Exactly, if opbeans are treated in the same manner, then we can ideally expect reproducibility. though, you have already found a couple of issues :)

Expectations:
- Tag based releases, and a docker image tagged with the agent version)
- Main always pointing to the latest release, and therefore  a docker image tagged `latest`
- Docker labels are also added to the docker images, to follow the labels standardisation.
- Tags are immutable, or sort of, so there is a kind of warranty, the opbeans@vx.y.z was tested and worked, while `latest` does not provide any warranty by default, it's always generated after any merge-commit.

I'm just giving my thoughts, no strong opinion if teams wanna diverge from the original proposal. It was intended to treat the opbeans as first citizens, so APM agent teams are the ultimate owners.

**Comment by trentm** [comment]
From discussion on the weekly Node.js APM agent call today, the preference is for "1. Get bump-version.sh to update package-lock.json" ... i.e. this PR should go ahead.  While decoupling the "agent release"-process and "opbeans update to latest agent and release"-process would be a good thing. That is too much work for now.

**Comment by trentm** [comment]
@astorm @v1v When you get a chance please (re-)review this. I think it is good to go in now. Thanks.

**Review by astorm — APPROVED** [review]

**Pull request merged**
